react/OrderItems.tsx

```diff
@@ -0,0 +1,112 @@
+import { adjust } from 'ramda'
+import React, {
+  createContext,
+  FunctionComponent,
+  ReactNode,
+  useContext,
+  useState,
+} from 'react'
+import { branch, renderComponent } from 'recompose'
+import { compose, graphql } from 'react-apollo'
+import { useOrderManager } from 'vtex.order-manager/OrderManager'
+
+import ItemList from './graphql/itemList.graphql'
+import UpdateItem from './graphql/updateItem.graphql'
+
+interface Context {
+  itemList: Item[]
+  updateItem: (index: number, quantity: number) => void
+  loading: boolean
+}
+
+interface OrderItemsProviderProps {
+  children: ReactNode
+  ItemListQuery: any
+  UpdateItem: any
+}
+
+const OrderItemsContext = createContext<Context | undefined>(undefined)
+
+const EmptyState: FunctionComponent = ({ children }: any) => {
+  const updateItem = async (_: number, __: number) => {}
+  return (
+    <OrderItemsContext.Provider
+      value={{ itemList: [], updateItem, loading: true }}
+    >
+      {children}
+    </OrderItemsContext.Provider>
+  )
+}
+
+let lastUpdateToken = { outdated: false }
```

Comment:
Why is that here? Why not use useRef? Keeping it here will lead to bugs if there are two instances of this provider

Reply:
Changed to a much better approach which is to use `OrderManager`'s queue status instead as you suggested.